spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/jsbridge/SCJavascriptBridgeAPI.java

```diff
@@ -492,8 +495,8 @@ public void onNext(Boolean connected) {
 
         SpatialConnect sc = SpatialConnect.getInstance();
         SCBackendService backendService = sc.getBackendService();
-        if (backendService != null) {
-            backendService
+        if (backendService != null && backendService instanceof SCMqttBackendService) {
+            ((SCMqttBackendService) backendService)
                     .connectedToBroker
                     .subscribeOn(Schedulers.io())
                     .subscribe(wrapper);
```

Comment:
This method probably should be named something more generic like `handleBackendConnectionStatus` and the action should do something similar as well `Actions.BACKENDSERVICE_MQTT_CONNECTED` to `Actions.BACKENDSERVICE_CONNECTED`